lib/puppet/util/windows/sid.rb

```diff
@@ -74,11 +74,13 @@ def name_to_principal(name, allow_unresolved = false)
         string_to_sid_ptr(name) do |sid_ptr|
           raw_sid_bytes = sid_ptr.read_array_of_uchar(get_length_sid(sid_ptr))
         end
-      rescue
+      rescue Puppet::Util::Windows::Error => e
```

Comment:
before this change, we were rescuing `StandardError`, is `Puppet::Util::Windows::Error` all that we need to rescue?

Reply:
It seems so to me, but just in case I'm wrong and to keep things futureproof, I can revert that since `Puppet::Util::Windows::Error` inherits `Puppet::Error` which inherits `RuntimeError` which in the end inherits `StandardError` and Ruby will rescue anything that inherits from `StandardError` if nothing is specified after the `rescue` keyword. Great catch! 🚀